include/libopencm3/stm32/l1/ri.h

```diff
@@ -0,0 +1,658 @@
+#define RI_ASCR1_GR11_3         (1 << 29)
+#define RI_ASCR1_GR11_4         (1 << 30)
+#define RI_ASCR1_SCM            (1 << 31)
+
+/**
+ * RI analog switches control register 2
+ *
+ * The RI_ASCR2 register is used to configure the analog switches of groups of
+ * I/Os not linked to the ADC. In this way, predefined groups of I/Os can be
+ * connected together.
+ */
+#define RI_ASCR2_GR10_1         (1 << 0)
+#define RI_ASCR2_GR10_2         (1 << 1)
+#define RI_ASCR2_GR10_3         (1 << 2)
+#define RI_ASCR2_GR10_4         (1 << 3)
+#define RI_ASCR2_GR6_1          (1 << 4)
+#define RI_ASCR2_GR6_2          (1 << 5)
+#define RI_ASCR2_GR5_1          (1 << 6)
+#define RI_ASCR2_GR5_2          (1 << 7)
+#define RI_ASCR2_GR5_3          (1 << 8)
+#define RI_ASCR2_GR4_1          (1 << 9)
+#define RI_ASCR2_GR4_2          (1 << 10)
+#define RI_ASCR2_GR4_3          (1 << 11)
+#define RI_ASCR2_GR3_3          (1 << 16)
+#define RI_ASCR2_GR3_4          (1 << 17)
+#define RI_ASCR2_GR3_5          (1 << 18)
+#define RI_ASCR2_GR9_3          (1 << 19)
+#define RI_ASCR2_GR9_4          (1 << 20)
+#define RI_ASCR2_GR2_3          (1 << 21)
+#define RI_ASCR2_GR2_4          (1 << 22)
+#define RI_ASCR2_GR2_5          (1 << 23)
+#define RI_ASCR2_GR7_5          (1 << 24)
+#define RI_ASCR2_GR7_6          (1 << 25)
+#define RI_ASCR2_GR7_7          (1 << 26)
+#define RI_ASCR2_GR6_3          (1 << 27)
+#define RI_ASCR2_GR6_4          (1 << 28)
+
+/**
+ * RI hysteresis control register 1
+ *
+ * The RI_HYSCR1 register is used to enable/disable the hysteresis of the input
+ * Schmitt trigger of ports A and B.
+ */
+#define RI_HYSCR1_PA0           (1 << 0)
+#define RI_HYSCR1_PA1           (1 << 1)
+#define RI_HYSCR1_PA2           (1 << 2)
+#define RI_HYSCR1_PA3           (1 << 3)
+#define RI_HYSCR1_PA4           (1 << 4)
+#define RI_HYSCR1_PA5           (1 << 5)
+#define RI_HYSCR1_PA6           (1 << 6)
+#define RI_HYSCR1_PA7           (1 << 7)
+#define RI_HYSCR1_PA8           (1 << 8)
+#define RI_HYSCR1_PA9           (1 << 9)
+#define RI_HYSCR1_PA10          (1 << 10)
+#define RI_HYSCR1_PA11          (1 << 11)
+#define RI_HYSCR1_PA12          (1 << 12)
+#define RI_HYSCR1_PA13          (1 << 13)
+#define RI_HYSCR1_PA14          (1 << 14)
+#define RI_HYSCR1_PA15          (1 << 15)
+#define RI_HYSCR1_PB0           (1 << 16)
+#define RI_HYSCR1_PB1           (1 << 17)
+#define RI_HYSCR1_PB2           (1 << 18)
+#define RI_HYSCR1_PB3           (1 << 19)
+#define RI_HYSCR1_PB4           (1 << 20)
+#define RI_HYSCR1_PB5           (1 << 21)
+#define RI_HYSCR1_PB6           (1 << 22)
+#define RI_HYSCR1_PB7           (1 << 23)
+#define RI_HYSCR1_PB8           (1 << 24)
+#define RI_HYSCR1_PB9           (1 << 25)
+#define RI_HYSCR1_PB10          (1 << 26)
+#define RI_HYSCR1_PB11          (1 << 27)
+#define RI_HYSCR1_PB12          (1 << 28)
+#define RI_HYSCR1_PB13          (1 << 29)
+#define RI_HYSCR1_PB14          (1 << 30)
+#define RI_HYSCR1_PB15          (1 << 31)
```

Comment:
As discussed before, GPIO0-GPIO15 can be used. I'll change that too to
~~~
#define RI_HYSCR1_PA(x)         (x)
#define RI_HYSCR1_PB(x)         (x << 16)
~~~